test/hotspot/jtreg/gc/shenandoah/generational/TestOldGrowthTriggers.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @summary Test that growth of old-gen triggers old-gen marking
+ * @requires vm.gc.Shenandoah
+ * @library /test/lib
+ * @run driver TestOldGrowthTriggers
+ */
+
+import java.util.*;
+import java.math.BigInteger;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class TestOldGrowthTriggers {
+
+  public static void do_old_allocations() {
+    // Expect most of the BigInteger entries placed into array to be promoted, and most will eventually become garbage within old
+
+    final int array_size = 128 * 1024;   // 128K entries
+    BigInteger array[] = new BigInteger[array_size];
+    Random r = new Random(46);
+
+    for (int i = 0; i < array_size; i++) {
+      array[i] = new BigInteger(128, r);
+    }
+
+    for (int refill_count = 0; refill_count < 8; refill_count++) {
+      System.out.print("Doing refill: ");
+      System.out.println(Integer.toString(refill_count));
+      // Each refill repopulates array_size randomly selected elements within array
+      for (int i = 0; i < array_size; i++) {
+        int replace_index = r.nextInt(array_size);
+        int derive_index = r.nextInt(array_size);
+        switch (i & 0x3) {
+          case 0:
```

Comment:
Intended to break after each case.  Let me fix this and adjust refill iterations.

Reply:
Done.  Testing now.